re_data_ui/src/utils/helpers.ts

```diff
@@ -104,6 +104,9 @@ export const appendToMapKey = (
 export const supportedResTypes = new Set(['source', 'model', 'seed']);
 
 export const generateModelId = (details: DbtNode | DbtSource): string => {
-  const { database, schema, name, identifier } = details;
-  return `${database}.${schema}.${identifier || name}`.toLowerCase();
+  const {
+    database, schema,
+  } = details;
+  const identifier = details.resource_type === 'source' ? (details as DbtSource).identifier : details.name;
```

Comment:
@dejii isn't it a different logic?
Meaning for source we also maybe using `name` we just try `identifier` first
?

Reply:
Right, working with a theory that `identifier` is usually present in `sources`. I will make a change not to assume that.